src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -158,23 +159,34 @@ class SidebarLinks extends React.Component {
         this.state = {
             currentlyViewedReportID: props.currentlyViewedReportID,
             orderedReports: [],
+            priorityMode: props.priorityMode,
             unreadReports: SidebarLinks.getUnreadReports(props.reports || {}),
         };
     }
 
     static getDerivedStateFromProps(nextProps, prevState) {
         const shouldReorder = SidebarLinks.shouldReorder(nextProps, prevState.orderedReports, prevState.currentlyViewedReportID, prevState.unreadReports);
+        const switchingPriorityModes = nextProps.priorityMode !== prevState.priorityMode;
+
+        // Pull the reports we want to show on the left hand nav
         const recentReports = SidebarLinks.getRecentReports(nextProps);
-        const orderedReports = shouldReorder
+
+        // Determine whether we want to re-order/sort the reports on the left hand nav
+        const orderedReports = shouldReorder || switchingPriorityModes
             ? recentReports
-            : _.map(prevState.orderedReports,
-                orderedReport => _.chain(recentReports)
-                    .filter(recentReport => orderedReport.reportID === recentReport.reportID)
-                    .first()
-                    .value());
+            : _.chain(prevState.orderedReports)
+
+            // To preserve the order of the conversations, we map over the previous state's order of reports.
+            // Then match and replace older reports with the newer report conversations from recentReports
+                .map(orderedReport => _.find(recentReports, recentReport => orderedReport.reportID === recentReport.reportID))
```

Comment:
So we are taking the previous reports and replacing it with the new report... but why would the report itself be any different? Not sure if I understand what we are trying to achieve with this step in general. 

Though I do understand what we are doing after this to clean up the `undefined` reports.

Reply:
So again, not super familiar with this area, but as you probably know each report object has info. Specifically for the draft messages. That report object's `hasDraftMessage` of the previous state will be set to false. The new list of reports pulled from `getRecentReports()` will have an updated list of report object, meaning if we were to start typing a message, the report object's `hasDraftMessage` key would be set to true.

Reply:
Ok so the because we are building a new set of options some things about those options will change. Makes sense, thanks.